File: typescript/src/1979-findGCD.ts
function findGCD(nums: number[]): number {
  const min: number = Math.min(...nums);
  const max: number = Math.max(...nums);
  // 如果相等，或者可以相除，返回最小值
  if (min === max || max % min === 0) {
    return min;
  }
  let i = 0;
  for (i = Math.floor(min / 2 + 1); i > 0; i--) {
    if (max % i === 0 && min % i === 0) {
      return i;
    }
  }
  return 1;
}

export {findGCD};

File: python/python3/1979-findGCD.py
import math

def findGCD(self, nums: List[int]) -> int:
    list_max = max(nums)
    list_min = min(nums)
    if list_max == list_min or list_max % list_min == 0:
        return list_min
    start = math.floor(list_min / 2 + 1)
    while start > 0:
        if list_max % start == 0 and list_min % start == 0:
            return int(start)
        start = start - 1
    return 1
